### ISSUE TITLE
ElasticsearchContainer does not expose the expected port

### ISSUE THREAD

**Issue by Macrame** [testcontainers/testcontainers-java#3103]
I used testcontainer for unit tests and hoped to expose the standard port 9200 and 9300.
```
public static ElasticsearchContainer elasticsearchContainer = new ElasticsearchContainer().withExposedPorts(9200, 9300);
...
        elasticsearchContainer.start();
        logger.debug("Elasticsearch container is running on {}.", elasticsearchContainer.getHttpHostAddress());
```
There was the log:
```
2020-08-17 08:53:38.605  INFO [,,,] 22780 --- [    Test worker] d.e.c.4.1]                               : Creating container for image: docker.elastic.co/elasticsearch/elasticsearch:6.4.1
2020-08-17 08:53:38.682  INFO [,,,] 22780 --- [    Test worker] o.t.utility.RegistryAuthLocator          : Credential helper/store (docker-credential-desktop) does not have credentials for docker.elastic.co
2020-08-17 08:53:38.725  INFO [,,,] 22780 --- [    Test worker] d.e.c.4.1]                               : Starting container with ID: e107af977195a900d58d692baef59ebf83246d48d42f560b36ad55dba1dc109e
2020-08-17 08:53:38.982  INFO [,,,] 22780 --- [    Test worker] d.e.c.4.1]                               : Container docker.elastic.co/elasticsearch/elasticsearch:6.4.1 is starting: e107af977195a900d58d692baef59ebf83246d48d42f560b36ad55dba1dc109e
2020-08-17 08:53:38.992  INFO [,,,] 22780 --- [    Test worker] o.t.c.wait.strategy.HttpWaitStrategy     : /admiring_dijkstra: Waiting for 120 seconds for URL: http://localhost:32811/
2020-08-17 08:53:43.479  WARN [,,,] 22780 --- [8mL1B63l2TaHg-1] o.s.a.r.l.SimpleMessageListenerContainer : Consumer raised exception, processing can restart if the connection factory supports it. Exception summary: org.springframework.amqp.AmqpConnectException: java.net.ConnectException: Connection refused: connect
2020-08-17 08:53:45.089  INFO [,,,] 22780 --- [    Test worker] d.e.c.4.1]                               : Container docker.elastic.co/elasticsearch/elasticsearch:6.4.1 started in PT6.484242S
2020-08-17 08:53:45.090 DEBUG [,,,] 22780 --- [    Test worker] c.i.g.m.s.t.c.TestConfiguration          : Elasticsearch container is running on localhost:32811.
```
And I checked the containers:
```
docker ps -a
CONTAINER ID        IMAGE                                                 COMMAND                  CREATED              STATUS              PORTS                                              NAMES
88914546c298        docker.elastic.co/elasticsearch/elasticsearch:6.4.1   "/usr/local/bin/dock…"   About a minute ago   Up 3 seconds        0.0.0.0:32824->9200/tcp, 0.0.0.0:32823->9300/tcp   trusting_shaw
7a7e124aadeb        testcontainersofficial/ryuk:0.3.0                     "/app"                   6 minutes ago        Up 6 minutes        0.0.0.0:32822->8080/tcp                            testcontainers-ryuk-732fc0e1-bdac-4fbe-b708-c16d8217d717
```
Finally, the unit test failed.
```
org.springframework.data.elasticsearch.client.NoReachableHostException: Host 'localhost:9200' not reachable. Cluster state is offline.
```
How to specify the correct mapped port? I hope it runs as `0.0.0.0:9200->9200/tcp, 0.0.0.0:9200->9300/tcp` instead of `0.0.0.0:32824->9200/tcp, 0.0.0.0:32823->9300/tcp`

**Comment by bsideup** [comment]
Hi @Macrame,

Please see https://www.testcontainers.org/features/networking/ and  https://bsideup.github.io/posts/testcontainers_fixed_ports/.

tl;dr: to avoid port conflicts and due to the host not always being `localhost`, the ports are randomized that we expose two methods, `getHost` + `getMappedPort` to obtain the actual values.

**Comment by bsideup** [comment]
Also see the Elasticsearch module's docs:
https://www.testcontainers.org/modules/elasticsearch/

**Comment by Macrame** [comment]
I don't want to obtain the actual values. I have to use 0.0.0.0:9200 in my scene.

**Comment by bsideup** [comment]
I see that you're using Spring Framework. It is very easy to wire the actual values in Spring:
https://spring.io/blog/2020/03/27/dynamicpropertysource-in-spring-framework-5-2-5-and-spring-boot-2-2-6

**Comment by Macrame** [comment]
Ok, got it. Thanks very much.
